### PR TITLE
Add some diagnostic items for Clippy

### PR DESCRIPTION
**Pull request by camsteffen**

### CODE REVIEW

**Comment by camsteffen** [comment]
Hmph. I don't know how to fix that...

**Comment by camsteffen** [comment]
Adding tests didn't help...

**Comment by camsteffen** [comment]
`not(bootstrap)` fixed it. Squashed it in.

**Comment by Mark-Simulacrum** [comment]
Hm, I'm not sure why you're seeing that error. Maybe someone more familiar with that infra in the compiler can give a better hint -- r? @oli-obk

File: library/std/src/path.rs [hunk 1]
```diff
@@ -1074,6 +1074,7 @@ impl FusedIterator for Ancestors<'_> {}
 // When attribute privacy is implemented, `PathBuf` should be annotated as `#[repr(transparent)]`.
 // Anyway, `PathBuf` representation and layout are considered implementation detail, are
 // not documented and must not be relied upon.
+#[cfg_attr(not(any(bootstrap, test)), rustc_diagnostic_item = "PathBuf")]
```
Comment (oli-obk) [inline library/std/src/path.rs:1077]: The reason these are warning about unused attributes is that the exact same attribute is already defined 10 lines above. I don't remember the implementation details of the `rustc_diagnostic_item` attribute, but I may have been lazy and not added a custom error for multiple attributes, and instead just looked for the first one and marked that as used.

Comment (camsteffen) [reply library/std/src/path.rs:1077]: Well...that's a little embarrassing. 🤦 But now I know to look for attributes above non-doc comments.

Comment (camsteffen) [reply library/std/src/path.rs:1077]: @oli-obk Would this be a good use for `delay_span_bug`?

**Comment by camsteffen** [comment]
Removed redundant Path/PathBuf attributes

File: library/std/src/path.rs (hunk 1 above)
Comment (oli-obk) [reply library/std/src/path.rs]: iirc there's a `find` somewhere that we should turn into a full for loop instead and report errors after the first one.

Comment (oli-obk) [reply library/std/src/path.rs]: I mean... technically the current warning is correct. it's just not very helpful

Comment (camsteffen) [reply library/std/src/path.rs]: Like this? https://github.com/camsteffen/rust/commit/dfe0399c574b976ac87e4dc4e64a1ac92e4fee99

**Comment by camsteffen** [comment]
CC @flip1995

**Pull request merged**